cmd/kubeadm/app/constants/constants.go

```diff
@@ -577,16 +577,6 @@ func GetAdminKubeConfigPath() string {
 	return filepath.Join(KubernetesDir, AdminKubeConfigFileName)
 }
 
-// GetSuperAdminKubeConfigPath returns the location on the disk where admin kubeconfig is located by default
-func GetSuperAdminKubeConfigPath() string {
-	return filepath.Join(KubernetesDir, SuperAdminKubeConfigFileName)
-}
-
-// GetBootstrapKubeletKubeConfigPath returns the location on the disk where bootstrap kubelet kubeconfig is located by default
-func GetBootstrapKubeletKubeConfigPath() string {
-	return filepath.Join(KubernetesDir, KubeletBootstrapKubeConfigFileName)
-}
-
```

Comment:
are there places where we should be calling these functions instead of constructing the paths there?

Reply:
Like these？
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go#L593
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/join/kubelet.go#L116
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go#L164-L166

Reply:
seems like all places that construct the super-admin.conf path use a dynamic dir variable.

so the default path location  `filepath.Join(constants.KubernetesDir, constants.SuperAdminKubeConfigFileName)` is never used. it always like `filepath.Join(dirVariable, constants.SuperAdminKubeConfigFileName)`

```
$ grep SuperAdminKubeConfigFileName ./cmd/kubeadm/* -rnI | grep KubernetesDir
./cmd/kubeadm/app/phases/upgrade/staticpods.go:517:             if _, err := os.Stat(filepath.Join(pathMgr.KubernetesDir(), constants.SuperAdminKubeConfigFileName)); err == nil {
./cmd/kubeadm/app/constants/constants.go:585:   return filepath.Join(KubernetesDir, SuperAdminKubeConfigFileName)
./cmd/kubeadm/app/constants/constants_test.go:54:       expected := filepath.Join(KubernetesDir, SuperAdminKubeConfigFileName)
```

it seems fine to remove this function and tests.

/lgtm
/approve